### PR TITLE
Add IsExecutable function to FileConventions

### PR DESCRIPTION
**Pull request by realmarv**

### CODE REVIEW

**Comment by knocte** [comment]
But we need a script that calls this, otherwise it's useless.

**Comment by realmarv** [comment]
> But we need a script that calls this, otherwise it's useless.

But how do I add a script when CI doesn't keep execution permission for the files

**Comment by knocte** [comment]
> But how do I add a script when CI doesn't keep execution permission for the files

I didn't mention CI at all, I said "add a script".

**Comment by knocte** [comment]
Let's rebase this PR (and fix CI) when you have time.

**Comment by realmarv** [comment]
> > But how do I add a script when CI doesn't keep execution permission for the files
> 
> I didn't mention CI at all, I said "add a script".

Done

**Comment by realmarv** [comment]
> Let's rebase this PR (and fix CI) when you have time.

Done

**Comment by knocte** [comment]
I added comments in the last 2 commits.

**Comment by realmarv** [comment]
> I added comments in the last 2 commits.

Done

**Comment by knocte** [comment]
Let's rebase this PR to get rid of the red CI.

**Comment by realmarv** [comment]
> Let's rebase this PR to get rid of the red CI.

Done

**Comment by knocte** [comment]
Why is CI red? Anyway this needs another rebase sorry.

File: scripts/executableConvention.fsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    )
+
+let invalidFiles =
+    Directory.GetFiles(".", "*.fsx", SearchOption.AllDirectories)
+    |> Seq.map(fun pathStr -> FileInfo pathStr)
+    |> Seq.filter(NotInDir "node_modules")
+    |> Seq.filter(NotInDir ".git")
+    |> Seq.filter(NotInDir "bin")
+    |> Seq.filter(NotInDir "obj")
+    |> Seq.filter(NotInDir "DummyFiles")
```
Comment (knocte) [inline scripts/executableConvention.fsx:22]: @realmarv I believe there's a new func in FileConventions that has these list above already, let's use it here

Comment (realmarv) [reply scripts/executableConvention.fsx]: Done

**Comment by knocte** [comment]
> But how do I add a script when CI doesn't keep execution permission for the files

I never understood well what you meant by this, so I tested it myself:

https://github.com/nblockchain/conventions/commits/wip/addIsExecutableFunctionToFileConventions

Conclusion: I don't know what kind of conundrum did your brain build, but this works as expected 😅

**Comment by realmarv** [comment]
> > But how do I add a script when CI doesn't keep execution permission for the files
> 
> I never understood well what you meant by this, so I tested it myself:
> 
> https://github.com/nblockchain/conventions/commits/wip/addIsExecutableFunctionToFileConventions
> 
> Conclusion: I don't know what kind of conundrum did your brain build, but this works as expected sweat_smile

If I recall correctly, I created a file with +x flag and printed that file's permissions in the CI and it didn't have +x flag.

**Comment by knocte** [comment]
> I created a file with +x flag and printed that file's permissions in the CI and it didn't have +x flag.

Well, you see it has worked on my build. So what you're saying is not true, you must have tested it wrongly.

**Comment by realmarv** [comment]
> > I created a file with +x flag and printed that file's permissions in the CI and it didn't have +x flag.
> 
> Well, you see it has worked on my build. So what you're saying is not true, you must have tested it wrongly.

I added a CI step to run the script. It's working as you said.

**Comment by knocte** [comment]
CI is still red. BTW, let's update the ReadMe.md file too.

**Comment by realmarv** [comment]
> CI is still red. BTW, let's update the ReadMe.md file too.

Done

**Comment by realmarv** [comment]
Rebased

**Pull request merged**